DDG4/src/Geant4ShapeConverter.cpp

```diff
@@ -274,6 +274,7 @@ namespace dd4hep {
         }
         g4->AddFacet(g4f);
       }
+      g4->SetSolidClosed(sh->IsClosedBody());
```

Comment:
I think I made a bug when implementing this and forgot the closure. Thanks a lot.
I have some hope that this fix will also resolve the issue:
https://github.com/AIDASoft/DD4hep/issues/879#issuecomment-1074165657

Guts feeling says this issue is connected, since Geant4 does not model assemblies as shapeless volumes, but places the daughters directly into the assembly mother...

Thanks for spotting this!